libs/log/default.go

```diff
@@ -0,0 +1,96 @@
+package log
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/rs/zerolog"
+)
+
+var _ Logger = (*DefaultLogger)(nil)
+
+// DefaultLogger implements a default logger that can be used within Tendermint
+// and that fulfills the Logger interface. The underlying logging provider is a
+// zerolog logger that supports typical log levels along with JSON and plain/text
+// log formats.
+//
+// Since zerolog supports typed structured logging and it is difficult to reflect
+// that in a generic interface, all logging methods accept a series of key/value
+// pair tuples, where the key must be a string.
+type DefaultLogger struct {
+	zerolog.Logger
+
+	trace bool
+}
+
+func NewDefaultLogger(format, level string, trace bool) (DefaultLogger, error) {
```

Comment:
can't this just return an object of our Logger's interface type and we can avoid exporting `DefaultLogger` as a type?

Reply:
Yes, but best to return concrete types and not interfaces where possible.

Reply:
That being said, I'm not hard-pressed on this. I mainly just wanted to remove all the existing logging cruft.